src/harness/unittests/cancellableLanguageServiceOperations.ts

```diff
@@ -0,0 +1,85 @@
+/// <reference path="..\harness.ts" />
+
+namespace ts {
+    describe("cancellableLanguageServiceOperations", () => {
+        const file = `
+        function foo(): void;
+        function foo<T>(x: T): T;
+        function foo<T>(x?: T): T | void {}
+        foo(f);
+        `;
+        it("can cancel signature help mid-request", () => {
+            verifyOperationCancelledAfter(file, 4, service => // Two calls are top-level in services, one is the root type, and the second should be for the parameter type
+                service.getSignatureHelpItems("file.ts", file.lastIndexOf("f"))

```

Comment:
I would also add check that after cancellation similar next request(without cancellation) will run correctly ?